src/ucai/core/utils/databricks_utils.py

```diff
@@ -0,0 +1,72 @@
+import logging
+from typing import TYPE_CHECKING
+
+_logger = logging.getLogger(__name__)
+
+if TYPE_CHECKING:
+    from databricks.sdk import WorkspaceClient
+
+
+def get_default_databricks_workspace_client() -> "WorkspaceClient":
+    try:
+        from databricks.sdk import WorkspaceClient
+
+        return WorkspaceClient()
+    except ImportError as e:
+        raise ImportError(
+            "Could not import databricks-sdk python package. "
+            "If you want to use databricks backend then "
+            "please install it with `pip install databricks-sdk`."
+        ) from e
+    except Exception as e:
+        _logger.warning(
+            "Failed to initialize databricks workspace client "
+            f"Falling back to NoOpClient. Error: {e}"
+        )
+    return NoOpClient()
+
+
+class NoOpStatementExecutionAPI:
+    def execute_statement(self, *args, **kwargs):
+        from databricks.sdk.service.sql import (
+            ResultData,
+            ResultManifest,
+            StatementResponse,
+            StatementState,
+            StatementStatus,
+        )
+
+        return StatementResponse(
+            manifest=ResultManifest(truncated=False),
+            result=ResultData(data_array=[["null"]]),
```

Comment:
can we pass an empty array to data_array?

Reply:
We can't as we need to find this field in execute_uc_function, and data_array is of List[List[str]] type.

Reply:
Emm we use `data_array[0][0]` to fetch the value, if the value is None, then normally the returned data_array would be `[["null"]]` as it's json serialized

Reply:
looks like we have a gaurd `data_array and len(data_array) > 0 and len(data_array[0]) > 0`:

https://github.com/serena-ruan/unitycatalog-ai/blob/ed7c0ba39be89b51e91bd8105115ecbaf2d5d030/src/ucai/core/databricks.py#L451-L455